hilti/toolchain/src/ast/types/function.cc

```diff
@@ -27,16 +27,28 @@ bool type::Function::isResolved(node::CycleDetector* cd) const {
     return true;
 }
 
-bool type::isValidOverload(Function* f1, Function* f2) {
+hilti::Result<Nothing> type::isValidOverload(Function* f1, Function* f2) {
+    // Equivalent functions that aren't hooks
+    if ( f1->flavor() != function::Flavor::Hook && areEquivalent(f1, f2) )
```

Comment:
The check for hook flavor doesn't seem right here and we might still work on hooks below. I'd suggest to instead move this into the caller, e.g., there

```cpp
if ( current_fn_ty->flavor() == type::function::Flavor::Hook &&
     current_fn_ty->flavor() == previous_fn_ty->flavor() )
    continue;
```

and here
```suggestion
    if ( areEquivalent(f1, f2) )
```

Reply:
I'll commit that as-suggested, but this makes me wonder if more thought should go into hooks. I think we could probably find errors there.

(ugh)